plugins/webhooks/index.js

```diff
@@ -253,7 +253,10 @@ async function triggeredPipelines(pipelineFactory, scmConfig, branch, type, acti
     let pipelinesOnOtherBranch = [];
 
     pipelines.forEach((p) => {
-        if (p.scmUri.startsWith(scmUri)) {
+        const tmpScmUri = p.scmUri.split(':');
+        const tmpScmBranch = `${tmpScmUri[0]}:${tmpScmUri[1]}:${tmpScmUri[2]}`;
+
+        if (tmpScmBranch === scmUri) {
```

Comment:
[ask] Why cannot we just say `p.scmUri === scmUri` ?
Is there any case `${tmpScmUri[3]}` ?

Reply:
This check is until `branch_name`.
because 'root_dir' suffix may have been appended, so `root_dir` excluded

`scmUri` structure is here 
`scm_uri_domain:repo_id:branch_name:root_dir`

Reply:
I see. You mean that `${tmpScmUri[3]}` will be the `root_dir`?

Reply:
yes

Reply:
And this comparing is not enough.
If `scmUri` is `github.com:1234:master:/somedir` and  `p.scmUri` is`github.com:1234:master`,  then this expression will be `false` and trigger with `pipelinesOnOtherBranch`.
So we should compare with `branch`. These pipelines must have same `scmRepoId`.